XbSymbolDatabase.h

```diff
@@ -551,6 +951,73 @@ void XbSymbolContinuousSigScan(bool enable);
+// before call XbSymbolContext_Create function.
+
+/// <summary>
+/// Step 1: Generate library array for LibraryHeader input.
+/// First call with <paramref name="library_out"/> as null pointer will return total count. Then second call will insert information to <paramref name="library_out"/>.filters field.
+/// </summary>
+/// <param name="xb_header_addr"></param>
+/// <param name="library_out">Input null pointer will provide the count to allocate memory for library_out.filters field. Otherwise, information will be input to library_out.filters field.</param>
+unsigned int XbSymbolDatabase_GenerateLibraryFilter(const void* xb_header_addr, XbSDBLibraryHeader* library_out);
+
+/// <summary>
+/// Step 2: Generate section array for SectionHeader input.
+/// First call with <paramref name="section_out"/> as null pointer will return total count. Then second call will insert information to <paramref name="section_out"/>.filters field.
+/// </summary>
+/// <param name="xb_header_addr"></param>
+/// <param name="section_out">Input null pointer will provide the count to allocate memory, then allocate memory to section_out.filters field.</param>
+/// <param name="is_raw">True: Convert array to use raw xbe's relative address; False: Convert array to use xbox virtual memory relative address.</param>
+unsigned int XbSymbolDatabase_GenerateSectionFilter(const void* xb_header_addr, XbSDBSectionHeader* section_out, bool is_raw);
+
+/// <summary>
+/// Step 3: Create XbSymbolContextHandle context for the scan process.
+/// </summary>
+/// <param name="ppHandle">Output XbSymbolContextHandle handler.</param>
+/// <param name="register_func">Register callback function for any detected symbols during scan process.</param>
+/// <param name="library_input">See XbSymbolDatabase_GenerateLibraryFilter for details.</param>
+/// <param name="section_input">See XbSymbolDatabase_GenerateSectionFilter for details.</param>
+/// <param name="kernel_thunk">Input decoded virtual kernel thunk.</param>
+bool XbSymbolContext_Create(XbSymbolContextHandle* ppHandle, xb_symbol_register_t register_func, XbSDBLibraryHeader library_input, XbSDBSectionHeader section_input, xbaddr kernel_thunk);
+
+/// <summary>
+/// Step 4: Perform a manual scan to set xreferences manually by requirement.
+/// </summary>
+/// <param name="pHandle">Input XbSymbolContextHandle handler.</param>
+void XbSymbolContext_ScanManual(XbSymbolContextHandle pHandle);
+
+/// <summary>
+/// Step 5: (multi-thread) Process individual library input by third-party.
+/// </summary>
+/// <param name="pHandle">Input XbSymbolContextHandle handler.</param>
+/// <param name="pLibrary">Input pointer of a library to start a scan process.</param>
+/// <param name="xref_first_pass">True: Optimized first search scan process; False: 2nd call and later will continue to return non-zero; see return for detail.</param>
+/// <returns>Return total xref count found. If return zero, then there's nothing left to find.</returns>
```

Comment:
> If return zero, then there's nothing left to find.

This isn't grammatically correct.
Should this be or something like this:

> If it returns zero, then there's nothing left to find.

Could be made even better with more changes.